proxy/src/control/cache.rs

```diff
@@ -54,37 +63,47 @@ impl<T> Cache<T> where T: Clone + Copy + Eq + std::hash::Hash {
     }
 
     pub fn extend<I, F>(&mut self, iter: I, on_change: &mut F)
-        where I: Iterator<Item = T>,
-              F: FnMut(T, CacheChange),
+    where
```

Comment:
I don't think `extend()` is a good name for this function any more. I suggest `extend_replacing_duplicates()`.